core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -550,19 +550,28 @@ private void saveCustomNameservers(Stack stack, KerberosConfig kerberosConfig, M
         }
     }
 
+    //TODO CB-11572 pipa
     private void saveDatalakeNameservers(Stack stack, Map<String, SaltPillarProperties> servicePillar) {
-        Long datalakeResourceId = stack.getDatalakeResourceId();
-        if (datalakeResourceId != null) {
-            Optional<DatalakeResources> datalakeResource = datalakeResourcesService.findById(datalakeResourceId);
-            if (datalakeResource.isPresent() && datalakeResource.get().getDatalakeStackId() != null) {
-                Long datalakeStackId = datalakeResource.get().getDatalakeStackId();
-                Stack dataLakeStack = stackService.getByIdWithListsInTransaction(datalakeStackId);
-                String datalakeDomain = dataLakeStack.getGatewayInstanceMetadata().get(0).getDomain();
-                List<String> ipList = dataLakeStack.getGatewayInstanceMetadata().stream().map(InstanceMetaData::getPrivateIp).collect(Collectors.toList());
-                servicePillar.put("forwarder-zones", new SaltPillarProperties("/unbound/forwarders.sls",
-                        singletonMap("forwarder-zones", singletonMap(datalakeDomain, singletonMap("nameservers", ipList)))));
-            }
+        Optional<Stack> datahubStack = datalakeService.getDatalakeStackByDatahubStack(stack);
```

Comment:
I suggest renaming it to `dataLakeStackOptional`.

Reply:
Ok